internal/apis/certmanager/types_certificaterequest.go

```diff
@@ -41,58 +48,88 @@ const (
 // configured issuers.
 //
 // All fields within the CertificateRequest's `spec` are immutable after creation.
-// A CertificateRequest will either succeed or fail, as denoted by its `status.state`
-// field.
+// A CertificateRequest will either succeed or fail, as denoted by its `Ready` status
+// condition and its `status.failureTime` field.
```

Comment:
**question:** Isn't the `Ready` condition also an indication of a failure?

Reply:
That is what the comment says right?